extensions/lifecycle/LifecycleMetrics.js

```diff
@@ -0,0 +1,89 @@
+const { ZenkoMetrics } = require('arsenal').metrics;
+
+const LIFECYCLE_LABEL_ORIGIN =  'origin';
+const LIFECYCLE_LABEL_PROCESS = 'process';
+const LIFECYCLE_LABEL_OP = 'op';
+const LIFECYCLE_LABEL_STATUS = 'status';
+
+const conductorLastBatchTime = ZenkoMetrics.createGauge({
+    name: 'lifecycle_conductor_last_batch_time',
+    help: 'Timestamp of latest lifecycle batch',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN
+    ],
+});
+
+const lifecycleS3Operations = ZenkoMetrics.createCounter({
+    name: 'lifecycle_s3_operations',
+    help: 'Total number of S3 operations by the lifecycle processes',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN,
+        LIFECYCLE_LABEL_OP,
+        LIFECYCLE_LABEL_STATUS,
+        LIFECYCLE_LABEL_PROCESS,
+    ],
+});
+
+const lifecycleKafkaPublish = ZenkoMetrics.createCounter({
+    name: 'lifecycle_kafka_publish',
+    help: 'Total number messages published by lifecycle processes',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN,
+        LIFECYCLE_LABEL_STATUS,
```

Comment:
Since status here is binary (success vs error), it could be 2 counters, as they are easy to sum, without the label tax.

Reply:
ok. I can split the counter into separate counts for the different http code ranges

Reply:
Here specifically for Kafka messages and only 2 outcomes possible it can be 2 counters. For HTTP status code ranges, a label is common practice and a reasonable use of cardinality.